File: 2019/day2/day2.pl
#!/usr/bin/perl
use strict;
use warnings;

sub run_program {
	my ($program) = @_;

	my $pos = 0;

	while (1) {
		my $opcode = $program->[$pos];
		if ($opcode == 99) {
			last;
		}
		my @args = @$program[$pos+1..$pos+3];
		if ($opcode == 1) {
			$program->[$args[2]] = $program->[$args[0]] + $program->[$args[1]];
		}
		elsif ($opcode == 2) {
			$program->[$args[2]] = $program->[$args[0]] * $program->[$args[1]];
		}
		$pos += 4;
	}
	return $program->[0];
}

open my $fh, '<', 'input.txt';
chomp(my $line = <$fh>);
close $fh;
my @program = split(/,/,$line);

$program[1] = 12;
$program[2] = 2;

print run_program(\@program) . "\n";

File: 2020/day2/day2.py
#!/usr/bin/python3

with open('input.txt') as f:
        lines = f.read().splitlines()

part1 = 0
part2 = 0
for line in lines:
	parts = line.split(" ")
	mn, mx = parts[0].split("-")
	mn = int(mn)
	mx = int(mx)
	letter = parts[1][0]
	count = 0
	for char in parts[2]:
		if char == letter:
			count += 1

	if mn <= count and count <= mx:
		part1 += 1

	count = 0
	for pos in (mn - 1, mx - 1):
		if parts[2][pos] == letter:
			count += 1

	if count == 1:
		part2 += 1
	
print(part1)
print(part2)
